packages/core/src/lib/history.integration.test.ts

```diff
@@ -0,0 +1,94 @@
+import { mkdir, rm, writeFile } from 'node:fs/promises';
+import { join } from 'node:path';
+import { type SimpleGit, simpleGit } from 'simple-git';
+import { afterAll, beforeAll, describe, expect } from 'vitest';
+import { getHashes } from './history';
+
+// we need a separate folder that is not cleaned in `global-setup.ts`, otherwise the tests can't execute in parallel
+const gitTestFolder = 'core-history-git-test';
+describe('git utils in a git repo', () => {
+  const baseDir = join(process.cwd(), gitTestFolder);
+  let emptyGit: SimpleGit;
+
+  beforeAll(async () => {
+    await mkdir(baseDir, { recursive: true });
+    emptyGit = simpleGit(baseDir);
+    await emptyGit.init();
+    await emptyGit.addConfig('user.name', 'John Doe');
+    await emptyGit.addConfig('user.email', 'john.doe@example.com');
+  });
+
+  afterAll(async () => {
+    await rm(baseDir, { recursive: true, force: true });
+  });
+
+  describe('without a branch and commits', () => {
+    it('getHashes should throw', async () => {
+      await expect(getHashes({}, emptyGit)).rejects.toThrow(
+        "your current branch 'master' does not have any commits yet",
+      );
+    });
+  });
+
+  describe('with a branch and commits clean', () => {
+    const commits: string[] = [];
+    beforeAll(async () => {
+      await writeFile(join(baseDir, 'README.md'), '# hello-world\n');
+      await emptyGit.add('README.md');
+      await emptyGit.commit('Create README');
+      // eslint-disable-next-line functional/immutable-data
+      commits.push((await emptyGit.log()).latest?.hash);
```

Comment:
TS is unhappy about this.
I would almost say it is okay to use `!` here because the commit happened in the previous command. WDYT? cc @matejchalk 

Applicable to all places where this happens.
```suggestion
      commits.push((await emptyGit.log()).latest!.hash);
```

Reply:
> I would almost say it is okay to use ! here because the commit happened in the previous command. WDYT? cc

Totally agree. The `no-non-null-assertion` rule is turned off in tests files for this reason.